ra/dsl.py

```diff
@@ -0,0 +1,419 @@
+import warnings
+import six
+import simplejson as json
+
+from . import raml
+from . import factory
+from .utils import path_from_uri, merge_query_params, path_to_identifier
+from .hooks import Hooks
+from .validate import RAMLValidator
+
+
+class APISuite(object):
+    """Represents an API test suite.
+    """
+    instances = []
+
+    def __new__(cls, *args, **kwargs):
+        instance = super(APISuite, cls).__new__(cls)
+        cls.instances.append(instance)
+        return instance
+
+    def __init__(self, raml_path_or_string, app, JSONEncoder=None):
+        """Instantiates an API test suite for the given RAML and app.
+        """
+        self.app = app
+        self.test_suite = TestSuite()
+
+        self.raml_path, self.raml = _parse_raml(raml_path_or_string)
+        self.path_prefix = path_from_uri(self.raml.base_uri)
+        self.resource_scopes = []
+
+        self.RequestClass = make_request_class(app, self.raml)
+
+        self.JSONEncoder = JSONEncoder or json.JSONEncoder
+
+        self.hooks = Hooks()
+        self.examples = self._define_factories()
+
+    def _define_factories(self):
+        examples = factory.Examples()
+        for resources in six.itervalues(self.raml.resources):
+            try:
+                post_resource = resources['POST']
+                example = post_resource.body['application/json'].example
```

Comment:
Will example factory (or other things related to using example) work properly if `post_resource.body['application/json'].example` is `None`?